spdif_rx.c

```diff
@@ -421,24 +420,26 @@ int spdif_rx_detect(spdif_rx_samp_freq_t *samp_freq, bool *inverted)
             if (r + bit_pos <= 31) { // within 32bit
                 succ += r;
                 int next = 1 - cur;
-                if (min_width[cur] > succ) {
-                    min_width[cur] = succ;
-                    // early termination by min_width (the higher frequency, the fewer samples for equivalent number of frames)
-                    if (min_width[cur] < SYSTEM_CLK_FREQUENCY / SAMP_FREQ_192000 / 128 - SC) {
-                        break; // no hope in this case, force termination
-                    } else if (min_width[cur] <= SYSTEM_CLK_FREQUENCY / SAMP_FREQ_176400 / 128 + SC) {
-                        num_check_words = SPDIF_RX_DETECT_SIZE / 4;
-                    } else if (min_width[cur] <= SYSTEM_CLK_FREQUENCY / SAMP_FREQ_88200 / 128 + SC) {
-                        num_check_words = SPDIF_RX_DETECT_SIZE / 2;
+                if (edge_pos[next]) { // valid previous edge?
```

Comment:
I had something like that, by initializing edge_pos[] to {-1, 0} and checking with if (edge_pos[next] >= 0). So the initial rising edge is already defined.
But then I got a doubt that the initial PIO edge detector might neglect to store one sample, because the check was done before the loop, so this would not be counted. Might be compensated by initializing succ to 1 instead of 0?
Instead of pulling my hair I chose to ignore that prior knowledge.

Reply:
thanks for the explanation. It makes sense about lost 1 cycle by `wait` instruction. Okay, don't make the code meaninglessly complex for the care of starting edge. I accept this.